packages/jest-cli/src/search_source.js

```diff
@@ -238,4 +239,30 @@ export default class SearchSource {
       return Promise.resolve({tests: []});
     }
   }
+
+  async getTestPaths(
+    globalConfig: GlobalConfig,
+    changedFilesPromise: ?ChangedFilesPromise,
+  ): Promise<SearchResult> {
+    const searchResult = await this._getTestPaths(
+      globalConfig,
+      changedFilesPromise,
+    );
+
+    if (globalConfig.filter) {
+      let tests = searchResult.tests;
+
+      // $FlowFixMe: dynamic require.
+      const filter = require(globalConfig.filter);
+      const filtered = await filter(tests.map(test => test.path));
+
+      if (Array.isArray(filtered)) {
+        tests = tests.filter(test => filtered.includes(test.path));
+      }
+
+      searchResult.tests = tests;
```

Comment:
Can we instead clone the object instead of mutating it?

```
Object.assign({}, searchResult, {tests});
```

Reply:
What's the reason for that?

Reply:
Because the `searchResult` object is received from another function, mutating it seems bad practice to me as there is no guarantee it isn't passed on to something else internally where this would cause trouble.

Silly example:

```
const getObject = () => {
  const object = {apple: {banana: 'kiwi'}};
  setTimeout(() => {
    console.log(object.apple.banana); // boom
   }, 100);
  return object;
};

getObject().apple = null;
```

Now, we shouldn't have cases like this in the codebase but if we do, this is super tough to debug. Let's avoid it.

Reply:
The search result object is created inside the function, so it can never be passed from the outside :)

Reply:
The variable (`searchResult`) is created in this function but the object is created in a different place in this class, thus making my original concern valid.

Reply:
What I mean is that the original `searchResult` will never be used anywhere else. The called method creates it and if I clone it, the original reference will get GC'ed as soon as you leave the method.